Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,164 @@
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The virtual bus is aimed at being a minimal software bus where virtbus
+devices carved out to represent a part of their core device's functionality can
+live on. This bus is also meant to be generic and be available to be used by
+any type of device (PCI/ACPI/DT). This means that virtbus devices created by
+different subsystems may co-exist on the same bus without sharing much in
+common with each other. Only the devices created by the same device share a
+common parent and establish a shared communication. The minimalist
+implementation for virtual bus also allows for extensibility by making
+specialized use of the virtbus_device by encapsulating it within other
+domain-specific structures.
+
+Virtbus Devices
+~~~~~~~~~~~~~~~
+
+A virtbus_device is created and registered to represent a part of its parent
+device's functionality. It is given a match_name that is used for driver
+binding and a release callback that is invoked when the device is unregistered.
+
+.. code-block:: c
+
+	struct virtbus_device {
+		struct device dev;
+		const char *match_name;
+		void (*release)(struct virtbus_device *);
+		u32 id;
+	};
+
+Virtbus Drivers
+~~~~~~~~~~~~~~~~~~~
+
+Virtbus drivers follow the standard driver model convention, where
+discovery/enumeration is handled outside the drivers, and drivers
+provide probe() and remove() methods. They support power management
+and shutdown notifications using the standard conventions.
+
+.. code-block:: c
+
+	struct virtbus_driver {
+		int (*probe)(struct virtbus_device *);
+		int (*remove)(struct virtbus_device *);
+		void (*shutdown)(struct virtbus_device *);
+		int (*suspend)(struct virtbus_device *, pm_message_t);
+		int (*resume)(struct virtbus_device *);
+		struct device_driver driver;
+		const struct virtbus_device_id *id_table;
+	};
+
+Virtbus drivers register themselves with the bus by calling
+virtbus_register_driver(). The id_table contains the names of virtbus devices
+that can bind with the driver.
+
+Device Enumeration
+~~~~~~~~~~~~~~~~~~
+
+The virtbus device is enumerated when it is attached to the bus. The device
+is assigned a unique ID automatically that will be appended to its name. If
+two virtbus_devices both named "foo" are registered onto the bus, they will
+have the device names, "foo.x" and "foo.y", where x and y are unique integers.
+
+Driving Binding
+~~~~~~~~~~~~~~~
+
+When a virtbus device is registered, the driver core iterates through the
+drivers registered with the virtual bus to identify a match between the device's
+match_name and the names in the driver's id_table. Once a successful match is
+found, the driver's probe is invoked binding the device to the driver upon
```

Comment:
the driver's probe is invoked, binding the driver.

'invoked binding' is unclear without comma I think.